test/extensions/filters/network/ext_authz/ext_authz_fuzz_test.cc

```diff
@@ -0,0 +1,125 @@
+#include "envoy/extensions/filters/network/ext_authz/v3/ext_authz.pb.h"
+
+#include "common/buffer/buffer_impl.h"
+#include "common/network/address_impl.h"
+
+#include "extensions/filters/network/ext_authz/ext_authz.h"
+
+#include "test/extensions/filters/common/ext_authz/mocks.h"
+#include "test/extensions/filters/network/ext_authz/ext_authz_fuzz.pb.validate.h"
+#include "test/fuzz/fuzz_runner.h"
+#include "test/mocks/network/mocks.h"
+#include "test/mocks/runtime/mocks.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using testing::ReturnRef;
+using testing::WithArgs;
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ExtAuthz {
+
+Filters::Common::ExtAuthz::ResponsePtr
+makeAuthzResponse(Filters::Common::ExtAuthz::CheckStatus status) {
+  Filters::Common::ExtAuthz::ResponsePtr response =
+      std::make_unique<Filters::Common::ExtAuthz::Response>();
+  response->status = status;
+  return response;
+}
+
+Filters::Common::ExtAuthz::CheckStatus resultCaseToCheckStatus(
+    envoy::extensions::filters::network::ext_authz::Result::ResultSelectorCase result_case) {
+  Filters::Common::ExtAuthz::CheckStatus check_status;
+  switch (result_case) {
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusOk: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::OK;
+    break;
+  }
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusError: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Error;
+    break;
+  }
+  case envoy::extensions::filters::network::ext_authz::Result::kCheckStatusDenied: {
+    check_status = Filters::Common::ExtAuthz::CheckStatus::Denied;
+    break;
+  }
+  default: {
+    // Unhandled status
+    PANIC("A check status handle is missing");
+  }
+  }
+  return check_status;
+}
+
+DEFINE_PROTO_FUZZER(const envoy::extensions::filters::network::ext_authz::ExtAuthzTestCase& input) {
+  try {
+    TestUtility::validate(input);
+  } catch (const ProtoValidationException& e) {
+    ENVOY_LOG_MISC(debug, "ProtoValidationException: {}", e.what());
+    return;
+  } catch (const ProtobufMessage::DeprecatedProtoFieldException& e) {
+    ENVOY_LOG_MISC(debug, "DeprecatedProtoFieldException: {}", e.what());
+    return;
+  } catch (const EnvoyException& e) {
```

Comment:
@jianwen612 This might need some clean-up. There is no return statement here so the fuzzer would continue down the below block of code. 

However, we never end up failing on this line anyway since the exceptions thrown from validate are caught in the previous two catches. We could/should unify all these catches and catch EnvoyExceptions, I'm not sure if we need to have 3 separate behaviors for the catches since they'll all lead to early return (and we except to catch EnvoyExceptions)

Reply:
> @jianwen612 This might need some clean-up. There is no return statement here so the fuzzer would continue down the below block of code.
> 
> However, we never end up failing on this line anyway since the exceptions thrown from validate are caught in the previous two catches. We could/should unify all these catches and catch EnvoyExceptions, I'm not sure if we need to have 3 separate behaviors for the catches since they'll all lead to early return (and we except to catch EnvoyExceptions)

Thanks for pointing this out! I will fix this the next week!